sopel/bot.py

```diff
@@ -660,37 +664,72 @@ def dispatch(self, pretrigger):
         for priority in ('high', 'medium', 'low'):
             items = self.get_triggered_callables(priority, pretrigger, blocked)
             for func, trigger, is_blocked in items:
+                function_name = "%s.%s" % (func.__module__, func.__name__)
                 # skip running blocked functions, but track them for logging
                 if is_blocked:
-                    function_name = "%s.%s" % (
-                        func.__module__, func.__name__
-                    )
                     list_of_blocked_functions.append(function_name)
                     continue
 
                 # call triggered function
                 wrapper = SopelWrapper(self, trigger)
                 if func.thread:
+                    # run in a separate thread
                     targs = (func, wrapper, trigger)
                     t = threading.Thread(target=self.call, args=targs)
+                    t.name = '%s-%s' % (t.name, function_name)
                     t.start()
+                    running_triggers.append(t)
                 else:
+                    # direct call
                     self.call(func, wrapper, trigger)
 
+        # log blocked functions
         if list_of_blocked_functions:
             if nick_blocked and host_blocked:
                 block_type = 'both'
             elif nick_blocked:
                 block_type = 'nick'
             else:
                 block_type = 'host'
-            LOGGER.info(
-                "[%s]%s prevented from using %s.",
+            LOGGER.debug(
+                "[%s] %s prevented from using %s.",
                 block_type,
                 pretrigger.nick,
                 ', '.join(list_of_blocked_functions)
             )
 
+        # update currently running triggers
+        self._update_running_triggers(running_triggers)
+
+    @property
+    def running_triggers(self):
+        """Current active threads for triggers.
+
+        This read-only list contains the currently running thread processing
+        a trigger. This is for testing and debugging purpose only.
+        """
+        with self._running_triggers_lock:
+            return [t for t in self._running_triggers if t.is_alive()]
+
+    def _update_running_triggers(self, running_triggers):
```

Comment:
This, too, could be named `__update_running_triggers` and thus completely inaccessible from outside this object, unless you need it to be accessible.

Reply:
Another trick of OOP & Python convention: `_` is for protected which means access is restricted to the class and its subclasses, while `__` is for private which means access is restricted to the class but **not** its subclasses - which makes it very inconvenient.